src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs

```diff
@@ -197,34 +197,43 @@ public void Start()
                 }
 
                 // STEP 2: Open the socket connection to the remote host
-                var cts = new CancellationTokenSource();
-                var tok = cts.Token;
-                
-                tok.Register(() =>
-                {
-                    _client.Close();
-                });
+                var cts = new CancellationTokenSource(ConnectTimeout);
```

Comment:
This constructor and the below `CancelAfter` are redundant, they both do the same thing.